### DISCUSSION TITLE
How frequent are ping() / is_valid() called in Connection

### DISCUSSION THREAD

**Discussion by wz-naquadah** [diesel-rs/diesel#3072] [Q&A] [answered]
Hi

I've been using rust/diesel to talk to cockroachdb. I've noticed one type of queries are way higher or more frequent from cockroachdb side

<img width="835" alt="image" src="https://user-images.githubusercontent.com/99854533/156900904-695db3a9-efaf-4e0c-87f4-c4a0356aeda3.png">

The `select _` query is basically a statement like `select 1`. I searched in the repo and found it's been used in the ping() method, which is used by is_valid() method.

I'm wondering where is_valid() method called since I can't search anything for it. I hope to find some clue/tips to reduce that query.

Thank you!

**Comment by weiznich** [comment, marked as answer]
This query is executed by the r2d2 connection pool every time you checkout an connection from the pool. It is used to check if the connection still works. 
You can reduce the number of times this query is executed by:
* Not using the connection pool
* Reduce the number of times you checkout a connection from the pool.

**Comment by wz-naquadah** [comment]
That is very help. Thank you! @weiznich
